### PR TITLE
Remove the exclamation mark in the proxy's global rule configuration #22329

### PR DESCRIPTION
**Pull request by jiangML**
For #22329.

Changes proposed in this pull request:
  - Remove the exclamation mark in the proxy's global rule configuration
  - Update doc
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/config/ProxyConfigurationLoader.java [hunk 1]
```diff
@@ -85,6 +88,18 @@ private static YamlProxyServerConfiguration loadServerConfiguration(final File y
         if (null != result.getAuthority()) {
             result.getRules().add(result.getAuthority().convertToYamlAuthorityRuleConfiguration());
         }
+        result.getRules().removeIf(each -> each instanceof YamlTransactionRuleConfiguration);
```
Comment (RaigorJiang) [inline proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/config/ProxyConfigurationLoader.java:91]: I suggest adding an interface `YamlGlobalRuleConfiguration`, which can help us remove duplicate code.

File: kernel/transaction/core/src/main/java/org/apache/shardingsphere/transaction/yaml/config/YamlTransactionConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    
+    private Properties props;
+    
+    @Override
+    public Class<? extends RuleConfiguration> getRuleConfigurationType() {
+        return TransactionRuleConfiguration.class;
+    }
+    
+    /**
+     * Convert to yaml transaction rule configuration.
```
Comment (RaigorJiang) [inline kernel/transaction/core/src/main/java/org/apache/shardingsphere/transaction/yaml/config/YamlTransactionConfiguration.java:47]: `yaml` should be `YAML`

File: proxy/bootstrap/src/main/resources/conf/server.yaml [hunk 3]
```diff
@@ -42,19 +42,19 @@
 #  privilege:
 #    type: ALL_PERMITTED
 #
-#rules:
-#- !TRANSACTION
+#transaction:
```
Comment (RaigorJiang) [inline proxy/bootstrap/src/main/resources/conf/server.yaml:45]: Please update the test configuration, too.
image

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/config/ProxyConfigurationLoader.java (hunk 1 above)
Comment (jiangML) [reply proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/config/ProxyConfigurationLoader.java]: ok

File: proxy/bootstrap/src/main/resources/conf/server.yaml (hunk 3 above)
Comment (jiangML) [reply proxy/bootstrap/src/main/resources/conf/server.yaml:45]: ok, i'll adjust it

File: kernel/transaction/core/src/main/java/org/apache/shardingsphere/transaction/yaml/config/YamlTransactionConfiguration.java (hunk 2 above)
Comment (jiangML) [reply kernel/transaction/core/src/main/java/org/apache/shardingsphere/transaction/yaml/config/YamlTransactionConfiguration.java]: ok, i'll adjust it

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/config/ProxyConfigurationLoader.java [hunk 4]
```diff
@@ -77,15 +77,28 @@ private static File getResourceFile(final String path) {
     
     private static YamlProxyServerConfiguration loadServerConfiguration(final File yamlFile) throws IOException {
         YamlProxyServerConfiguration result = YamlEngine.unmarshal(yamlFile, YamlProxyServerConfiguration.class);
-        if (null == result) {
-            return new YamlProxyServerConfiguration();
-        }
         // TODO authority will no longer be a global rule
```
Comment (RaigorJiang) [inline proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/config/ProxyConfigurationLoader.java:80]: Please remove this comment.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**